stack/moving-to-github-actions.rst

```diff
@@ -0,0 +1,76 @@
+
+Travis uses a ``.travis.yml`` file that specifies the workflow to be executed when events such as pushes or pull requests occur for the repository.
+GitHub Actions use ``.github/workflows/*.yaml`` files instead.
+These workflows execute in parallel.
+Migrating a package consists of replacing the ``.travis.yml`` file with one (or more) appropriate GitHub Actions workflows files.
+
+If the ``.travis.yml`` file contains only the default execution of ``flake8`` for Python "lint" checking, this can be done entirely in :ref:`the GitHub web interface <github-actions-web-interface>`, or it can be done a bit more cleanly from :ref:`the command line <github-actions-command-line>`.
+
+If your ``.travis.yml`` does more than run ``flake8``, please consult an expert for assistance.
+
+
+.. _github-actions-web-interface:
+
+GitHub web interface
+====================
+
+#. Go to the repository you are transitioning at ``https://github.com/lsst/{repo}``.
+#. Select ``.travis.yml``
+#. Select the "trash can" icon to delete this file.
+#. Add a commit comment such as "Remove Travis workflow." and choose a branch name.
+   This branch name doesn't need to follow our ``tickets/`` convention; ``u/{user}/remove-travis`` is adequate.
+#. Commit the change, create a pull request, and merge it.
+#. Select the "Actions" tab between "Pull Requests" and "Projects" at the top of the GitHub page.
+#. Choose the "LSST DM Python lint Workflow" and click on "Set up this workflow".
+#. Choose "Start commit".
+#. Add a commit comment such as "Add Python lint workflow." and choose a branch name.
+   Again, something like ``u/{user}/add-lint-workflow`` is sufficient as a branch name.
+#. Commit the change, create a pull request, and merge it.
+#. Update the repository settings using :ref:`the procedure below <github-actions-settings>`.
+
+
+.. _github-actions-command-line:
+
+Command line
+============
+
+#. Clone the repository you are transitioning from ``git@github.com:lsst/{repo}``.
+#. Create a branch: ``git checkout -b u/{user}/migrate-to-gha``
+#. ``git rm .travis.yml``
+#. ``mkdir -p .github/workflows; cd .github/workflows``
+#. ``curl -LO https://https://raw.githubusercontent.com/lsst/.github/master/workflow-templates/lint.yaml``
+#. ``git add lint.yaml``
+#. ``git commit`` and add a commit message such as "Migrate from Travis to GitHub Actions."
+#. ``git push -u origin u/{user}/migrate-to-gha``
+#. Create a pull request and merge it.
+#. Update the repository settings using :ref:`the procedure below <github-actions-settings>`.
+
+
+.. _github-actions-settings:
+
+Settings changes
+================
+
+#. Go to the Settings page of the repository.
+   This may be under the right-side "..." menu depending on your window width.
+#. Choose "Branches" in the left sidebar.
+#. Under "Branch protection rules", click "Edit" next to branch "master".
+#. Under "Requre branches to be up to date before merging", uncheck amy status checks that contain "travis-ci".
```

Comment:
```suggestion
#. Under "Require branches to be up to date before merging", uncheck any status checks that contain "travis-ci".
```

Reply:
@andy-slac asked about the travis webhooks -- I assume we ignore them?

Reply:
When I've moved projects I haven't gone out of my way to delete those webhooks; Travis seems to ignore any webhook event for repos that don't have a `.travis.yml`. I'd only worry about it if you want really tidy repo settings.